File: exercises/week02/di-scala/src/decoupled/StandardOutMessageRenderer.scala
package decoupled

class StandardOutMessageRenderer {
  private var messageProvider: HelloWorldMessageProvider = null

  def render {
    if (messageProvider == null) {
      throw new RuntimeException("You must set the property messageProvider of class:" + classOf[StandardOutMessageRenderer].getName)
    }
    println(messageProvider.getMessage)
  }

  def getMessageProvider: HelloWorldMessageProvider = {
    return messageProvider
  }

  def setMessageProvider(provider: HelloWorldMessageProvider) {
    messageProvider = provider
  }
}

File: exercises/week02/di/src/decoupled/StandardOutMessageRenderer.java
package decoupled;

public class StandardOutMessageRenderer {

    private HelloWorldMessageProvider messageProvider = null;

    public void render() {
        if (messageProvider == null) {
            throw new RuntimeException(
                    "You must set the property messageProvider of class:"
                            + StandardOutMessageRenderer.class.getName());
        }
        System.out.println(messageProvider.getMessage());
    }

    public HelloWorldMessageProvider getMessageProvider() {
        return this.messageProvider;
    }

    public void setMessageProvider(HelloWorldMessageProvider provider) {
        this.messageProvider = provider;
    }

}